network/src/sync/blocks/base.rs

```diff
@@ -0,0 +1,147 @@
+// Copyright (C) 2019-2021 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// The snarkOS library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+use std::{net::SocketAddr, time::Duration};
+
+use futures::{pin_mut, select, FutureExt};
+use snarkos_storage::Digest;
+use snarkvm_dpc::BlockHeaderHash;
+use tokio::{sync::mpsc, time::Instant};
+
+use crate::{Node, Peer, SyncInbound};
+use anyhow::*;
+
+pub struct SyncBase {
+    pub node: Node,
+    pub incoming: mpsc::Receiver<SyncInbound>,
+}
+
+impl SyncBase {
+    pub fn new(node: Node) -> (Self, mpsc::Sender<SyncInbound>) {
+        let (sender, receiver) = mpsc::channel(256);
+        let new = Self {
+            node,
+            incoming: receiver,
+        };
+        (new, sender)
+    }
+
+    pub async fn find_sync_nodes(&self) -> Result<Vec<Peer>> {
+        let our_block_height = self.node.storage.canon().await?.block_height;
+        let mut interesting_peers = vec![];
+        for mut node in self.node.peer_book.connected_peers_snapshot().await {
+            let judge_bad = node.judge_bad();
+            if !judge_bad && node.quality.block_height as usize > our_block_height + 1 {
+                interesting_peers.push(node);
+            }
+        }
+        interesting_peers.sort_by(|x, y| y.quality.block_height.cmp(&x.quality.block_height));
```

Comment:
```suggestion
        interesting_peers.sort_by_key(|x| std::cmp::Reverse(x.quality.block_height));
```
`clippy` will suggest this

Reply:
`clippy` didn't suggest, avoiding for now.